File: splearn-java/src/test/java/io/wisoft/splearn/SplearnTestConfiguration.java
package io.wisoft.splearn;

import io.wisoft.splearn.application.required.EmailSender;
import io.wisoft.splearn.domain.MemberFixture;
import io.wisoft.splearn.domain.PasswordEncoder;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;

@TestConfiguration
public class SplearnTestConfiguration {
    @Bean
    public EmailSender emailSender() {
        return (email, subject, body) -> System.out.println("Sending email: " + email);
    }

    @Bean
    public PasswordEncoder passwordEncoder() {
        return MemberFixture.createPasswordEncoder();
    }
}

File: splearn-typescript/src/test/splearn-test-configuration.ts
import { Email } from "@src/main/domain/shared/email";
import { createPasswordEncoder } from "@src/test/domain/member/member.fixture";

import type { PasswordEncoder } from "@src/main/domain/member/password-encoder";

export class SplearnTestConfiguration {
  emailSender() {
    return {
      send: (
        email: Email,
        subject: string,
        body: string,
      ): void => {
        console.info(`To: ${ email.address }, Subject: ${ subject }, Body: ${ body }`);
      },
    };
  }

  passwordEncoder(): PasswordEncoder {
    return createPasswordEncoder();
  }
}